src/basic/time-util.c

```diff
@@ -1604,15 +1605,72 @@ bool clock_supported(clockid_t clock) {
         }
 }
 
+int find_localtime_last_symlink(char **ret) {
+        _cleanup_free_ char *path = NULL, *last_symlink = NULL;
+        int r;
+        unsigned i;
+
+        assert(ret);
+
+        path = strdup("/etc/localtime");
+        if (!path)
+                return -ENOMEM;
+
+        last_symlink = strdup(path);
+        if (!last_symlink)
+                return -ENOMEM;
+
+        for (i = 0; i < CHASE_MAX; ++i) {
+                _cleanup_free_ char *next = NULL;
+                struct stat st;
+
+                r = lstat(path, &st);
```

Comment:
in new code, let's not do things by path anymore. pin the inode via an fd, and then go from inode to inode. chaseat() should help you with that.